src/coreclr/vm/typedesc.h

```diff
@@ -439,22 +439,35 @@ class TypeVarTypeDesc : public TypeDesc
 /*************************************************************************/
 /* represents a function type.  */
 
-typedef SPTR(class FnPtrTypeDesc) PTR_FnPtrTypeDesc;
+typedef DPTR(class FnPtrTypeDesc) PTR_FnPtrTypeDesc;
 
 class FnPtrTypeDesc : public TypeDesc
 {
 
 public:
 #ifndef DACCESS_COMPILE
-    FnPtrTypeDesc(BYTE callConv, DWORD numArgs, TypeHandle * retAndArgTypes)
-        : TypeDesc(ELEMENT_TYPE_FNPTR), m_NumArgs(numArgs), m_CallConv(callConv)
+    FnPtrTypeDesc(PTR_Module module, PCOR_SIGNATURE sig, int32_t sigLen, BYTE callConv, DWORD numArgs, TypeHandle * retAndArgTypes)
+        : TypeDesc(ELEMENT_TYPE_FNPTR), m_pModule(module), m_hExposedClassObject(0), m_NumArgs(numArgs), m_CallConv(callConv), m_sig(sig), m_sigLen(sigLen)
     {
         LIMITED_METHOD_CONTRACT;
+
+        m_sig = (PCOR_SIGNATURE)new uint8_t[sigLen];
+        memcpy(m_sig, sig, sigLen);
+
         for (DWORD i = 0; i <= numArgs; i++)
         {
             m_RetAndArgTypes[i] = retAndArgTypes[i];
         }
     }
+
+    ~FnPtrTypeDesc() {
```

Comment:
The type loader structrures cannot have destructors. The destructors won't run when the type is unloaded.

Instead, all memory has to be allocated using loader heaps. The whole loader heap gets destroyed when the type is unloaded.

Reply:
Thanks